pkg/clusterstatus/apiserver/apiserver.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	kubeoneapi "github.com/kubermatic/kubeone/pkg/apis/kubeone"
+	"github.com/kubermatic/kubeone/pkg/ssh"
+)
+
+const (
+	defaultHTTPTimeout = 10 * time.Second
+	healthzEndpoint    = "https://%s:6443/healthz"
+)
+
+type Status struct {
+	Health bool `json:"health,omitempty"`
+}
+
+type HTTPDoer interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+// CheckAPIServer uses the /healthz endpoint to check are all API server instances healthy
+func GetStatus(node kubeoneapi.HostConfig, tunneler ssh.Tunneler) (*Status, error) {
+	client := httpClient(tunneler)
+	health, err := apiserverHealth(client, node.PrivateAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Status{
+		Health: health,
+	}, nil
+}
+
+// apiserverHealth checks is API server healthy
+func apiserverHealth(c HTTPDoer, nodeAddress string) (bool, error) {
+	endpoint := fmt.Sprintf(healthzEndpoint, nodeAddress)
+	request, err := http.NewRequest("GET", endpoint, nil)
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := c.Do(request)
+	if err != nil {
+		return false, nil
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, err
+	}
+
+	return string(body) == "ok", nil
+}
+
+// httpClient builds an HTTP client used to access the API server
+func httpClient(tunneler ssh.Tunneler) HTTPDoer {
+	transport := &http.Transport{
+		DialContext: tunneler.TunnelTo,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
```

Comment:
For some reason, this endpoint only works with `InsecureSkipVerify` set to `true`. If I provide certs I get an error saying `Unauthorized` that looks like:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "Unauthorized",
  "reason": "Unauthorized",
  "code": 401
}
```

Reply:
@xmudrii probably because you've tried to use etcd client keys to access apiserver endpoint :)

Reply:
No, I have used the API server keys. Here is the `curl` command that also fails:
```
sudo curl --cacert /etc/kubernetes/pki/ca.crt --cert /etc/kubernetes/pki/apiserver.crt --key /etc/kubernetes/pki/apiserver.key https://<endpoint>:6443/healthz
```

Reply:
this is server cert, but you need client